README.md

```diff
@@ -1 +1,292 @@
+```
+
+The resulting module and main file will produce be able to switch between different settings according to the environment variable:
+
+```
+$ node main-environment
+URL is http://webapplog.com
+API key is D1FCFE58-575F-40FF-AF36-0CBD6A367756
+$ NODE_ENV=production node main-environment
+URL is http://webapplog.com
+API key is 34EC5CE9-CD02-4C4E-B5D9-D7CEF2F12F2C
+$ NODE_ENV=testing node main-environment
+URL is http://webapplog.com
+API key is D2230FC4-7E2A-4003-9372-0BF80107558D
+```
+
+To summarize the difference between objects and functions, the latter are more dynamic and allow for the so called initializer or constructor code, i.e., some code to tweak an object before exporting it.
+
+## module.exports and exports
+
+Simply put, `module.exports` is an alias to the `exports`. You can save your self a repetitive stress injury by using `exports`. Take a look at this module which uses `module.exports`:
+
+```js
+module.exports.sayHelloInEnglish = function() {
+  return 'Hello'
+}
+
+module.exports.sayHelloInTatar = function() {
+  return 'Isänmesez'
+}
+
+module.exports.sayHelloInSpanish = function() {
+  return 'Hola'
+}
+
+module.exports.sayHelloInJapanese = function() {
+  return 'こんにちは' // Kon'nichiwa
+}
+```
+
+The following version is an absolute exact equivalent:
+
+```js
+exports.sayHelloInEnglish = function() {
+  return 'Hello'
+}
+
+exports.sayHelloInTatar = function() {
+  return 'Isänmesez'
+}
+
+exports.sayHelloInSpanish = function() {
+  return 'Hola'
+}
+
+exports.sayHelloInJapanese = function() {
+  return 'こんにちは' // Kon'nichiwa
+}
+```
+
+That's great! In this case `exports` and `module.exports` are the same. What if we want to assign the whole object like we did with the configuration module to the `exports` object?
+
+So this is the working code with `module.exports`. Obviously the code with `module.exports` works, because we used it in the previous section:
+
+```js
+module.exports = function(env) {
+  env = env || process.env.NODE_ENV
+  if (env == 'production') {
+    ...
+  } else if (env == 'testing') {
+    ...
+  } else { // Assume development
+    ...
+  }
+}
+```
+
+What about this code with `exports`? Will it work too?
+
+```js
+exports = function(env) {
+  env = env || process.env.NODE_ENV
+  if (env == 'production') {
+    ...
+  } else if (env == 'testing') {
+    ...
+  } else { // Assume development
+    ...
+  }
+}
+```
+
+Go ahead and try to run it (`environment-exports.js`) with this one-liner command:
+
+```
+node -e "console.log(require('./environment-exports')().apiKey)"
+```
+
+If you're getting `TypeError: require(...) is not a function`, that's right. You are getting an error because now nothing has been exported. By writing `exports = ...`, we **lost** the original value of `exports`!
+
+Note: We lose the reference, because we overwrite the object `exports`. It's okay to overwrite `module.exports` because in this case `module.exports` is a property of the `module` object.
```

Comment:
This could be tough for a student to grasp. `module.exports` is what we get from a call to `require`; if we overwrite `exports`, we are overwriting the module's only means of exporting its functionality.[1]

[1]: http://www.commonjs.org/specs/modules/1.0/